pandas/io/json/_json.py

```diff
@@ -716,6 +719,12 @@ def _get_data_from_filepath(self, filepath_or_buffer):
                 errors=self.encoding_errors,
             )
             filepath_or_buffer = self.handles.handle
+        elif (
+            isinstance(filepath_or_buffer, str)
+            and filepath_or_buffer.lower().endswith((".json", ".json.gz", ".json.bz2"))
```

Comment:
We have a list of all supported compression extensions:
```py
from pandas.io.common import _compression_to_extension

.endswith([".json"] +  [f".json.{compression}" for compression in _compression_to_extension])
```

Reply:
Oh good, I was looking for that.